test/Constraints/closures.swift

```diff
@@ -349,13 +349,12 @@ takeVoidVoidFn { () -> Void in
 }
 
 // <rdar://problem/19997471> Swift: Incorrect compile error when calling a function inside a closure
-func f19997471(_ x: String) {}
-func f19997471(_ x: Int) {}
+func f19997471(_ x: String) {} // expected-note {{candidate expects value of type 'String' at position #0}}
+func f19997471(_ x: Int) {}    // expected-note {{candidate expects value of type 'Int' at position #0}}
 
 func someGeneric19997471<T>(_ x: T) {
   takeVoidVoidFn {
-    f19997471(x) // expected-error {{cannot invoke 'f19997471' with an argument list of type '(T)'}}
-    // expected-note @-1 {{overloads for 'f19997471' exist with these partially matching parameter lists: (Int), (String)}}
+    f19997471(x) // expected-error {{no exact matches in call to global function 'f19997471'}}
```

Comment:
Doesn't to be this pull request, but it would be really nice if we reminded the user of the types of the arguments here.

Reply:
Do you think it would be better to attach arguments to this diagnostic or to each note?